cmake/SetupDoxygen.cmake

```diff
@@ -14,25 +14,64 @@ if (DOXYGEN_FOUND)
   # Configure file that contains doxygen groups
   configure_file(docs/GroupDefs.hpp ${SPECTRE_DOXYGEN_GROUPS})
 
-  add_custom_target(
-      doc
-      COMMAND ${DOXYGEN_EXECUTABLE} ${PROJECT_BINARY_DIR}/docs/DoxyfileHtml 2>&1
-      DEPENDS
-      ${PROJECT_BINARY_DIR}/docs/DoxyfileHtml
-      ${SPECTRE_DOXYGEN_GROUPS}
-      )
-
-  add_custom_target(
-    doc-check
-    # search for warnings and print out 6 lines after to make sure
-    # majority of warning is printed so the results are actually
-    # useful.
-    COMMAND ! ${DOXYGEN_EXECUTABLE} ${PROJECT_BINARY_DIR}/docs/DoxyfileHtml 2>&1
-    | grep -A 6 'warning'
-    DEPENDS
-    ${PROJECT_BINARY_DIR}/docs/DoxyfileHtml
-    ${SPECTRE_DOXYGEN_GROUPS}
-    )
+  # We need Python for postprocessing the documentation
+  find_package(PythonInterp)
+  include(FindPythonModule)
+  find_python_module(bs4 FALSE)
+  find_python_module(pybtex FALSE)
+
+  if (PYTHONINTERP_FOUND AND PY_BS4 AND PY_PYBTEX)
+    add_custom_target(
+        doc
+        COMMAND
+        ${DOXYGEN_EXECUTABLE} ${PROJECT_BINARY_DIR}/docs/DoxyfileHtml 2>&1
+        && ${PYTHON_EXECUTABLE}
+        ${CMAKE_SOURCE_DIR}/docs/config/postprocess_docs.py -v
+        --html-dir ${PROJECT_BINARY_DIR}/docs/html
+        --references-file ${CMAKE_SOURCE_DIR}/docs/References.bib
+        DEPENDS
+        ${PROJECT_BINARY_DIR}/docs/DoxyfileHtml
+        ${SPECTRE_DOXYGEN_GROUPS}
+        )
+    add_custom_target(
+        doc-check
+        # search for warnings and print out 6 lines after to make sure
+        # majority of warning is printed so the results are actually
+        # useful.
+        COMMAND
+        ! ${DOXYGEN_EXECUTABLE} ${PROJECT_BINARY_DIR}/docs/DoxyfileHtml 2>&1
+        | grep -A 6 'warning'
+        && ${PYTHON_EXECUTABLE}
+        ${CMAKE_SOURCE_DIR}/docs/config/postprocess_docs.py
+        --html-dir ${PROJECT_BINARY_DIR}/docs/html
+        --references-file ${CMAKE_SOURCE_DIR}/docs/References.bib
+        DEPENDS
+        ${PROJECT_BINARY_DIR}/docs/DoxyfileHtml
+        ${SPECTRE_DOXYGEN_GROUPS}
+        )
+  else (PYTHONINTERP_FOUND AND PY_BS4 AND PY_PYBTEX)
+    message(WARNING "Doxygen documentation postprocessing is disabled because Python dependencies were not found.")
```

Comment:
Could you print which dependencies weren't found and maybe even how to install them? I don't think most people (including myself) will be familiarly with these dependencies :)